src/instructlab/train/linux_train.py

```diff
@@ -194,15 +194,17 @@ def linux_train(
     torch_dtype = "auto" if device.type == "cuda" else None
     if device.type == "cpu":
         total_memory = psutil.virtual_memory().total / (1024**3)
-        if total_memory < 64:
+        if total_memory < 60:
             # Using our default model, a system with 32 GB of RAM
             # will get OOM killed using torch_dtype=None, though we
             # seem to get much better performance with this setting
-            # where there's enough memory.
+            # where there's enough memory. Using `None` makes it
+            # use float32 as opposed to float16 or bf16.
             #
-            # There's more going on here and needs deeper exploration to find
-            # the right parameters to be checking for choosing the best
-            # configuration.
+            # Anecdotally, 64 GB seems to be enough, but this calculation
```

Comment:
Since it's such a rough guess, 60 still seems fine? We need to actually do some math at some point ...

Reply:
I'll share my math in a few :) stay tuned!

Reply:
![Screenshot 2024-06-07 at 14 23 04](https://github.com/instructlab/instructlab/assets/912735/19849592-dc9b-441c-b708-86b60a3af96d)

Some more numbers:

* The training part take ~30GB of RAM to process, there is a very small chance that this could work on very minimal Linux installation, by minimal I mean, only system critical services run and nothing else.
* The inference part takes ~35GB of RAM

Essentially a system with 48GB of RAM should be able to run both training and inferencing. Although 48 GB of RAM is not very common.